plugins/gui/src/WindowCollection.h

```diff
@@ -33,106 +41,99 @@ namespace gui {
-                    , fpsms_show_options(false)
-                    , fpsms_buffer_size(20)
-                    , fpsms_refresh_rate(2.0f)
-                    , fpsms_mode(TIMINGMODE_FPS)
-                    , buf_current_delay(0.0f)
-                    , buf_ms_values()
-                    , buf_fps_values()
-                    , buf_ms_max(1.0f)
-                    , buf_fps_max(1.0f)
-                    , tfe_view_minimized(false)
-                    , tfe_view_vertical(false)
-                    , tfe_active_param("")
-                    , buf_tfe_reset(false)
-                    , log_level(static_cast<int>(megamol::core::utility::log::Log::LEVEL_ALL))
-                    , log_force_open(true) {
-            }
+            unsigned int log_level = static_cast<int>(megamol::core::utility::log::Log::LEVEL_ALL); // Log level used in log window
+            bool log_force_open = true;             // flag indicating if log window should be forced open on warnings and errors
         };
 
+        struct Complete {
+            Basic basic;
+            Specific specific;
+        };
 
-        // --------------------------------------------------------------------
-        // WINDOWs
+        /** Type for predefined window callback function. */
+        typedef std::function<void(WindowConfiguration&)> PredefinedCallbackFunc_t;
 
-        WindowCollection() = default;
+        /** Type for unknown window callback function. */
+        typedef std::function<void(Basic&)> CallbackFunc_t;
 
+        WindowConfiguration(const std::string& name, PredefinedCallbackID cb_id) : hash_id(std::hash<std::string>()(name))
+                , name(name), callback_id(cb_id), volatile_callback(nullptr), config() {
+        }
+
+        WindowConfiguration(const std::string& name, CallbackFunc_t* cbf) : hash_id(std::hash<std::string>()(name))
+                , name(name), callback_id(DRAWCALLBACK_UNKNOWN), volatile_callback(cbf), config() {
+        }
+
+        ~WindowConfiguration(void) = default;
+
+        void SetVolatileCallback(CallbackFunc_t* callback) {
+            this->volatile_callback = callback;
+            this->callback_id = DRAWCALLBACK_UNKNOWN;
+        }
+
+        std::string Name(void) const { return this->name; }
+        size_t Hash(void) const { return this->hash_id; }
+        PredefinedCallbackID CallbackID(void) const { return this->callback_id; }
+        CallbackFunc_t* Callback(void) const { return this->volatile_callback; }
+        
+        void ApplyWindowSizePosition(bool consider_menu);
+        
+        Complete config;
+
+    private:
+
+        size_t hash_id;                    // unique hash generated from name to omit string comparison
+        std::string name;                  // unique name of the window
+        PredefinedCallbackID callback_id;  // ID of the predefined callback drawing the window content
+        CallbackFunc_t* volatile_callback; // [NOT SAVED] Alternative unknown window callback.
```

Comment:
I am wondering: can we make the `CallbackFunc_t` owned here? Then Services requesting GUI windows could just push a lambda callback without holding a member variable for that callback themselves. This way we only need to include `GUIWindowRequest.h` in the Service cpp file. Also, then the GUI implementation could wrap its own `PredefinedCallbackID` callbacks in `CallbackFunc_t` via function pointers and this would simplify the code calling/rendering these callbacks?